scenes/worlds/world_gen/generation_steps/generation_group.gd

```diff
@@ -0,0 +1,49 @@
+# Acts as a Group of GenerationSteps, it will execute all of it's children before moving on
+class_name GenerationGroup
+extends GenerationStep
+
+### Member Variables and Dependencies -------------------------------------------------------------
+#--- signals --------------------------------------------------------------------------------------
+
+#--- enums ----------------------------------------------------------------------------------------
+
+#--- constants ------------------------------------------------------------------------------------
+
+#--- public variables - order: export > normal var > onready --------------------------------------
+
+#--- private variables - order: export > normal var > onready -------------------------------------
+
+### -----------------------------------------------------------------------------------------------
+
+
+### Built-in Virtual Overrides --------------------------------------------------------------------
+
+### -----------------------------------------------------------------------------------------------
+
+
+### Public Methods --------------------------------------------------------------------------------
+
+func execute_step(data : WorldData, gen_data : Dictionary, generation_seed : int):
```

Comment:
Please avoid overriding function not marked for that unless strictly necessary, to avoid having to manually keep behavior in sync when changing it on the base class (for example, if GenerationStep starts measuring its own time and emitting the step finished signals itself, we would have to also update this here), and to avoid surprising behaviors when calling this function without realizing it might have been overriden. In this case, since it seems to do exactly the same as the base class, just removing the override definition should be enough.

Reply:
This is here because GenerationGroup didn't extend GenerationStep at the beggining, but later when I changed it forgot to erase it. Will erase it.

I'm not sure GenerationGroup should inherit GenerationStep, but due to some limitations in GDScript compared to GDscript 2.0 it was easier this way.

Reply:
Yeah, inheriting GenerationStep is probably the right choice in this case, makes it more modular. since for the caller, there won't be any difference, and we can have any depth of generation steps without issue.